pkg/reconciler/ingress/controller.go

```diff
@@ -168,3 +189,35 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
 
 	return impl
 }
+
+func waitForCache(caches *generator.Caches) {
+	timeout := time.After(cacheWarmUPTimeout)
+	tick := time.Tick(1 * time.Second)
+pollCacheInSync:
+	for {
+		select {
+		case <-timeout:
+			break pollCacheInSync
+		case <-tick:
+			if caches.HasSynced() {
+				break pollCacheInSync
+			}
+		}
+	}
+}
+
+func getReadyIngresses(knativeClient v1alpha12.NetworkingV1alpha1Interface) (map[string]struct{}, error) {
+	ingresses, err := knativeClient.Ingresses("").List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	ingressesToWarm := map[string]struct{}{}
+	for _, ingress := range ingresses.Items {
+		if val, ok := ingress.Annotations[networking.IngressClassAnnotationKey]; ok {
+			if val == config.KourierIngressClassName && ingress.GetStatus().GetCondition(v1alpha1.IngressConditionNetworkConfigured).IsTrue() {
```

Comment:
```suggestion
		if ingress.Annotations[networking.IngressClassAnnotationKey] == config.KourierIngressClassName && ingress.GetStatus().GetCondition(v1alpha1.IngressConditionNetworkConfigured).IsTrue() {
```

Has the same semantics I think.